iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformers.java

```diff
@@ -0,0 +1,475 @@
+        controlledRegulatingTerminals.replaceRegulatingTerminal(getTerminal1(twoR.t2w3, isWellOrientedT2w3), t3w.getLeg3().getTerminal());
+
+        // t3w is not considered in controlledRegulatingTerminals (created in the model later)
+        Map<Terminal, Terminal> regulatingTerminalMapping = getRegulatingTerminalMapping(twoR, t3w);
+        replaceRegulatingTerminal(t3w.getLeg1(), regulatingTerminalMapping);
+        replaceRegulatingTerminal(t3w.getLeg2(), regulatingTerminalMapping);
+        replaceRegulatingTerminal(t3w.getLeg3(), regulatingTerminalMapping);
+
+        copyTerminalActiveAndReactivePower(getTerminal1(twoR.t2w1, isWellOrientedT2w1), t3w.getLeg1().getTerminal());
+        copyTerminalActiveAndReactivePower(getTerminal1(twoR.t2w2, isWellOrientedT2w2), t3w.getLeg2().getTerminal());
+        copyTerminalActiveAndReactivePower(getTerminal1(twoR.t2w3, isWellOrientedT2w3), t3w.getLeg3().getTerminal());
+
+        copyStarBusVoltageAndAngle(twoR.starBus, t3w);
+        List<PropertyR> lostProperties = new ArrayList<>();
+        lostProperties.addAll(copyProperties(twoR.t2w1, t3w));
+        lostProperties.addAll(copyProperties(twoR.t2w2, t3w));
+        lostProperties.addAll(copyProperties(twoR.t2w3, t3w));
+
+        List<ExtensionR> lostExtensions = copyExtensions(twoR, t3w);
+
+        // copy necessary data before removing
+        List<AliasR> t2wAliases = new ArrayList<>();
+        t2wAliases.addAll(getAliases(twoR.t2w1, "1", getEnd1(isWellOrientedT2w1)));
+        t2wAliases.addAll(getAliases(twoR.t2w2, "2", getEnd1(isWellOrientedT2w2)));
+        t2wAliases.addAll(getAliases(twoR.t2w3, "3", getEnd1(isWellOrientedT2w3)));
+
+        String t2w1Id = twoR.t2w1.getId();
+        String t2w2Id = twoR.t2w2.getId();
+        String t2w3Id = twoR.t2w3.getId();
+        String starVoltageId = twoR.starBus.getVoltageLevel().getId();
+
+        remove(twoR);
+
+        // after removing
+        List<AliasR> lostAliases = copyAliases(t2wAliases, t3w);
+
+        // warnings
+        if (!lostProperties.isEmpty()) {
+            lostProperties.forEach(propertyR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + propertyR.t2wId + "' unexpected property '" + propertyR.propertyName + "'"));
+        }
+        if (!lostExtensions.isEmpty()) {
+            lostExtensions.forEach(extensionR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + extensionR.t2wId + "' unexpected extension '" + extensionR.extensionName + '"'));
+        }
+        if (!lostAliases.isEmpty()) {
+            lostAliases.forEach(aliasR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + aliasR.t2wId + "' unexpected alias '" + aliasR.alias + "' '" + aliasR.aliasType + "'"));
+        }
+
+        // report
+        ReportNode reportNodeReplacement = reportNode.newReportNode().withMessageTemplate("replaced-3t2w-by-t3w", "Replaced 3 TwoWindingsTransformers by ThreeWindingsTransformer").add();
```

Comment:
I moved the report code to a private method in the same class to avoid moving the record definitions to the ModificationReports class. Let me know If you still prefer moving all this code to the ModificationReports class.

Reply:
Done